Inferences/InvalidAnswerLiteralRemovals.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+/**
+ * @file InvalidAnswerLiteralRemoval.cpp
+ * Implements class UncomputableAnswerLiteralRemoval.
+ */
+
+#include "Kernel/Clause.hpp"
+#include "Parse/TPTP.hpp"
+
+#include "InvalidAnswerLiteralRemovals.hpp"
+
+namespace Inferences
+{
+
+Clause* UncomputableAnswerLiteralRemoval::simplify(Clause* cl)
+{
+  unsigned cLen = cl->length();
+  for (unsigned li = 0; li < cLen; li++) {
+    Literal* lit = (*cl)[li];
+    if (lit->isAnswerLiteral() && !lit->computableOrVar())
+      return nullptr;
+  }
+  return cl;
+}
+
+UndesiredAnswerLiteralRemoval::UndesiredAnswerLiteralRemoval(const vstring& avoidThese)
+{
+  // TODO: catch parsing exceptions and complain properly
+  _avoiders = Parse::TPTP::parseClauseFromString(avoidThese);
```

Comment:
I have to ask...what's going on here, and why are we parsing clauses?

Reply:
You can say: `./vampire -qa plain -qaat "ans0(sK1)|ans0(f(X))"` and we will skip proofs that would lead to answers matching `ans0(sK1)` or `ans0(f(X))`. It's quite advantageous to do this from the command line and not from a tptp  file (we can wait until the skolem makes sense) and it's quite advantageous to (ab)use the TPTP parser.

Reply:
Ahah. Neat! I failed to do the mental wiring-up, thanks.